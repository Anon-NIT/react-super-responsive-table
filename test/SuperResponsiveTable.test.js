import React from 'react'
import { Table, Thead, Tbody, Tr, Th, Td } from 'react-super-responsive-table'

test('Render Table', () => {
  const wrapper = shallow(
    <Table>
      <Thead>
        <Tr>
          <Th>Annual Conference</Th>
          <Th>Year</Th>
          <Th>Location</Th>
          <Th>President</Th>
          <Th>Program Chair</Th>
          <Th>Conference Theme</Th>
        </Tr>
      </Thead>
      <Tbody>
        <Tr>
          <Td>31</Td>
          <Td>2017</Td>
          <Td>Alabama Community College System (ACCS)</Td>
          <Td>Mr. Toner Evans, Samford University</Td>
          <Td>Ms. Kelly Birchfield, Auburn University Montgomery</Td>
        </Tr>
      </Tbody>
    </Table>
  )
  expect(wrapper).toMatchSnapshot()
})
