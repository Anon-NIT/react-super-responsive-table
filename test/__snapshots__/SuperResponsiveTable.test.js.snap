// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Table 1`] = `
<ContextProvider
  value={Object {}}
>
  <table
    className=" responsiveTable"
  >
    <Thead>
      <Tr>
        <Th>
          Annual Conference
        </Th>
        <Th>
          Year
        </Th>
        <Th>
          Location
        </Th>
        <Th>
          President
        </Th>
        <Th>
          Program Chair
        </Th>
        <Th>
          Conference Theme
        </Th>
      </Tr>
    </Thead>
    <Tbody>
      <Tr>
        <Td>
          31
        </Td>
        <Td>
          2017
        </Td>
        <Td>
          Alabama Community College System (ACCS)
        </Td>
        <Td>
          Mr. Toner Evans, Samford University
        </Td>
        <Td>
          Ms. Kelly Birchfield, Auburn University Montgomery
        </Td>
      </Tr>
    </Tbody>
  </table>
</ContextProvider>
`;

exports[`Render table with an only one column 1`] = `
<ContextProvider
  value={Object {}}
>
  <table
    className=" responsiveTable"
  >
    <Thead>
      <Tr>
        <Th>
          Annual Conference
        </Th>
      </Tr>
    </Thead>
    <Tbody>
      <Tr>
        <Td>
          31
        </Td>
      </Tr>
    </Tbody>
  </table>
</ContextProvider>
`;

exports[`Render table without any column 1`] = `
<ContextProvider
  value={Object {}}
>
  <table
    className=" responsiveTable"
  >
    <Thead>
      <Tr />
    </Thead>
    <Tbody>
      <Tr />
    </Tbody>
  </table>
</ContextProvider>
`;

exports[`Table updates correctly upon change of key prop: headersA 1`] = `
<Parent
  headers={
    Array [
      "alpha",
      "beta",
    ]
  }
  i={1}
>
  <Table
    key="1"
  >
    <table
      className=" responsiveTable"
    >
      <Thead>
        <thead>
          <Tr
            inHeader={true}
          >
            <TrInner
              headers={
                Object {
                  "0": "alpha",
                  "1": "beta",
                }
              }
              inHeader={true}
            >
              <tr>
                <Th
                  columnKey={0}
                  key="0/.0"
                >
                  <th>
                    alpha
                  </th>
                </Th>
                <Th
                  columnKey={1}
                  key="1/.1"
                >
                  <th>
                    beta
                  </th>
                </Th>
              </tr>
            </TrInner>
          </Tr>
        </thead>
      </Thead>
      <Tbody>
        <tbody>
          <Tr>
            <TrInner
              headers={
                Object {
                  "0": "alpha",
                  "1": "beta",
                }
              }
            >
              <tr>
                <Td
                  columnKey={0}
                  key="0/.0"
                >
                  <TdInner
                    columnKey={0}
                    headers={
                      Object {
                        "0": "alpha",
                        "1": "beta",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        alpha
                      </div>
                      item 1
                    </td>
                  </TdInner>
                </Td>
                <Td
                  columnKey={1}
                  key="1/.1"
                >
                  <TdInner
                    columnKey={1}
                    headers={
                      Object {
                        "0": "alpha",
                        "1": "beta",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        beta
                      </div>
                      item 2
                    </td>
                  </TdInner>
                </Td>
              </tr>
            </TrInner>
          </Tr>
        </tbody>
      </Tbody>
    </table>
  </Table>
</Parent>
`;

exports[`Table updates correctly upon change of key prop: headersB 1`] = `
<Parent
  headers={
    Array [
      "one",
      "two",
    ]
  }
  i={2}
>
  <Table
    key="2"
  >
    <table
      className=" responsiveTable"
    >
      <Thead>
        <thead>
          <Tr
            inHeader={true}
          >
            <TrInner
              headers={
                Object {
                  "0": "one",
                  "1": "two",
                }
              }
              inHeader={true}
            >
              <tr>
                <Th
                  columnKey={0}
                  key="0/.0"
                >
                  <th>
                    one
                  </th>
                </Th>
                <Th
                  columnKey={1}
                  key="1/.1"
                >
                  <th>
                    two
                  </th>
                </Th>
              </tr>
            </TrInner>
          </Tr>
        </thead>
      </Thead>
      <Tbody>
        <tbody>
          <Tr>
            <TrInner
              headers={
                Object {
                  "0": "one",
                  "1": "two",
                }
              }
            >
              <tr>
                <Td
                  columnKey={0}
                  key="0/.0"
                >
                  <TdInner
                    columnKey={0}
                    headers={
                      Object {
                        "0": "one",
                        "1": "two",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        one
                      </div>
                      item 1
                    </td>
                  </TdInner>
                </Td>
                <Td
                  columnKey={1}
                  key="1/.1"
                >
                  <TdInner
                    columnKey={1}
                    headers={
                      Object {
                        "0": "one",
                        "1": "two",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        two
                      </div>
                      item 2
                    </td>
                  </TdInner>
                </Td>
              </tr>
            </TrInner>
          </Tr>
        </tbody>
      </Tbody>
    </table>
  </Table>
</Parent>
`;
